packages/debug/src/browser/debug-configuration-manager.ts

```diff
@@ -254,21 +277,27 @@ export class DebugConfigurationManager {
 
     protected async doOpen(model: DebugConfigurationModel): Promise<EditorWidget> {
         let uri = model.uri;
-        if (!uri) {
+        const configFileUri = await this.getFolderConfigurationUri(model);
+        if (!await this.filesystem.exists(configFileUri.toString())) {
             uri = await this.doCreate(model);
         }
-        return this.editorManager.open(uri, {
+        return this.editorManager.open(uri!, {
```

Comment:
`!` why is it needed?, `if` branch should ensure that uri exists

Reply:
i agree with you while the linter doesn't. 

without `!` it gives me this error 
```
Argument of type 'URI | undefined' is not assignable to parameter of type 'URI'.
  Type 'undefined' is not assignable to type 'URI'.ts(2345)
```
and

```
   'Exit with failure status (1): concurrently -n compile,lint -c blue,green "theiaext compile" "theiaext lint"\nerror Command failed with exit code 1.\n',
```

Reply:
We should sort out other comments, I don't think that so many changes to `doCreate` function are required. It seems there are regressions with PreferenceService.